core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1330,7 +1330,7 @@ object TestUtils extends Logging {
     waitUntilTrue(() => {
       records ++= consumer.poll(50).asScala
       records.size >= numMessages
-    }, s"Consumed ${records.size} records until timeout instead of the expected $numMessages records", waitTime)
+    }, s"Could not consume $numMessages records before timing out", waitTime)
```

Comment:
Good point. We should make the string a by name parameter so that it is evaluated only when used. But it can be done separately if that's easier.